glib/src/value.rs

```diff
@@ -655,6 +655,20 @@ macro_rules! from_glib {
             }
         }
 
+        impl FromGlibPtrBorrow<*mut gobject_ffi::GValue> for $name {
+            #[inline]
+            unsafe fn from_glib_borrow(ptr: *mut gobject_ffi::GValue) -> Borrowed<$name> {
+                $crate::translate::from_glib_borrow::<_, $name>(ptr as *mut gobject_ffi::GValue)
```

Comment:
```suggestion
                Borrowed::new(ptr::read(ptr as *const $name))
```

Might work, but it's all a bit ugly.

Ideally in your callback what you'd do instead (manually, not via gir... but I guess gir could get support for that) would be:

```
unsafe extern "C" your_trampoline(..., value: *const GValue, ...) {
    your_rust_function(..., &*(value as *const glib::Value), ...)
}
```

Reply:
>  (manually, not via gir... but I guess gir could get support for that)

we have a bunch of functions that were manually implemented for this reason.  I would suggest that we rather fix this in gir directly

Reply:
I've been playing with GValue-related stuff and come to the conclusion `gir` just isn't producing the right code

Reply:
Yes it doesn't do that correctly in many cases with `GValue`s. Your best bet for now would be to bind the function manually. If you can give me a link to your attempt I'd be happy to review that.